internal/api/api.go

```diff
@@ -114,6 +114,12 @@ func NewAPIWithVersion(ctx context.Context, globalConfig *conf.GlobalConfigurati
 			}).SetBurst(30),
 		)).With(api.verifyCaptcha).Post("/token", api.Token)
 
+		r.With(api.limitHandler(
+			tollbooth.NewLimiter(api.config.RateLimitTokenRefresh/(60*5), &limiter.ExpirableOptions{
+				DefaultExpirationTTL: time.Hour,
+			}).SetBurst(30),
+		)).With(api.verifyCaptcha).Post("/oauth/token", api.OAuthPKCE)
```

Comment:
I also feel like this whole flow should go into the already existing `/token` endpoint. I know it's a bit overloaded, but I would rather not have 2 different `/token` endpoints that just confuse us or users.

Also, could you please make sure you add any new / changed APIs to the `openapi.yaml` file. You can use the [new Swagger editor](https://editor-next.swagger.io) to make the changes.

Reply:
Hmm i feel like this should be a separate endpoint because the `/token` endpoint is already confusing enough with 3 different grant_types we support: `password`, `refresh_token` and `id_token`. The `/oauth/token` endpoint is also only called subsequently, and not at the start of the auth flow.